src/Parser.php

```diff
@@ -1856,7 +1860,7 @@ private function parseExpressionFn() {
 
     /**
      * @param Node $parentNode
-     * @return Expression
+     * @return UnaryExpression|MissingToken|Variable
```

Comment:
Added `Variable` here because the `CastExpression->operand` accepts only a `Variable` (which is wrong but I guess will be caught on higher PHPStan levels).

I ran the tests to see if this function returns `Varaible` and it does.